Fix gizmo to cursor snap

IhaveJuss opened the pull request:
Fixes an unpleasant snap of the object to the cursor when we start position change. 

https://github.com/user-attachments/assets/d678600b-14ca-47dc-9540-388d520ac0ba

1. Fixes #96 
2. Additionally a change of how drag of different axis is handled, should work practically the same with a lot less repetition, can move this part to the separate PR if necessary.

Commit:
Fix gizmo to cursor snap

Fixes an unpleasant snap of the object to the cursor when we start
position change.

BlakeDarrow commented:
This looks great! Thank you again for your work! One small inconsistency. It appears that when the gizmo drag ends, sometimes the entity gets deselected. Previously, you could grab any axis or plane handle, transform it, and immediately do it again. It appears this might change that. The entity seems to become deselected if you end the drag over a mesh of the gizmo? Are you seeing similar behavior?

(Left is your branch, right is current dev)

https://github.com/user-attachments/assets/4760af7f-8d45-4c12-97b8-5fd7c512c3d2

IhaveJuss commented:
@BlakeDarrow Good catch, will try to reproduce later today.

Commit:
Fix entity deselect on drag end.

Looks like right now in bevy picking this sequence of
events:
1. `Pointer<Press>`
2. `Pointer<DragStart>`
...
3. `Pointer<DragEnd>`

Will provide a valid `Pointer<Click>` event. Which in turn for us
triggered unselect of selected object.

Validating only `original_event_target` instead of all pierced objects
gives this function ability to only know about click on the gizmo, which
will not trigger unselect.

IhaveJuss commented:
Should work as expected now, because we were taking into account pointer offset the `Press` and the `Release`  Pointer events was happening to the same object, which triggered `Pointer<Click>` again. 
Similar issue described here: https://github.com/bevyengine/bevy/issues/19833

As for our case we only concerned if the user clicked on intractable object or gizmos, simple change to `original_event_target` should fix the issue.
